### PR TITLE
Locale aware uppercase lowercase fix

### PR DESCRIPTION
**Pull request by gheja**
The original version uses `toLowerCase()` and `toUpperCase()` that are not locale aware, see Issue #239

For example the letter "i" is different in Turkish.
```
> 'i'.toUpperCase()
< 'I'

> 'i'.toLocaleUpperCase('tr-TR')
< 'İ'

> 'i'.toUpperCase().charCodeAt(0)
< 73

> 'i'.toLocaleUpperCase('tr-TR').charCodeAt(0)
< 304
```

I added a setting `LOCALE_STRING` that (if set) makes the code use `toLocaleLowerCase()` and `toLocaleUpperCase()` with the correct locale.

Thanks for the help and testing @Cryptoseen and @joedel263!

### CODE REVIEW

File: src/constants/settings.ts [hunk 1]
```diff
@@ -3,3 +3,4 @@ export const MAX_CHALLENGES = 6
 export const ALERT_TIME_MS = 2000
 export const REVEAL_TIME_MS = 350
 export const GAME_LOST_INFO_DELAY = (MAX_WORD_LENGTH + 1) * REVEAL_TIME_MS
+export const LOCALE_STRING = ''
```
Comment (cwackerfuss) [inline src/constants/settings.ts:6]: Thoughts on putting this in `.env` as `REACT_APP_LOCALE_STRING`?

Think it makes more sense than in settings.ts

**Review by cwackerfuss — CHANGES_REQUESTED** [review]
Looks good! Just one piece of feedback -- I think it makes more sense in .env

File: src/constants/settings.ts (hunk 1 above)
Comment (gheja) [reply src/constants/settings.ts:6]: Sounds reasonable, I will move it there.

Comment (gheja) [reply src/constants/settings.ts]: Moved it to .env.

**Review by cwackerfuss — APPROVED** [review]

**Pull request merged**